{% extends 'layout.html' %}
{% block error_notices %}
{% if error %}
<div class="alert alert-error">
  <button type="button" class="close" data-dismiss="alert">&times;</button>
  <strong>Error:</strong> {{error}}
</div>
{% endif %}
{% endblock %}
{% block body %}
<ul class="nav nav-tabs">
  <li class="active"><a href="#nucl" data-toggle="tab">Nucleotide input</a></li>
  <li><a href="#prot" data-toggle="tab">Protein input</a></li>
  <li><a href="#job" data-toggle="tab">Results for existing job</a></li>
</ul>
<div class="tab-content">
  <div class="tab-pane active" id="nucl">
    <div>Search a genome sequence for secondary metabolite biosynthesis gene clusters:</div>
    <div>
      <div class="control-group">
        <button class="btn btn-inverse" id="dna-sample-input">Load sample input</button>
        <button class="btn btn-inverse" id="dna-sample-output">Open example output</button>
      </div>
      <form class="form-vertical" method="POST" id="nucl-form" enctype="multipart/form-data">
        <fieldset class="as-form">
          <div class="control-group">
            <div class="controls">
              <input type="email" name="email" id="email" value="{{old_email}}" placeholder="your.email@example.com">
              <span class="help-inline">Email address (optional)</span>
            </div>
          </div>
          <div class="control-group">
            <div class="controls">
              <input type="file" name="seq" id="seq">
              <span class="help-inline">Load a file in GenBank / EMBL format (recommended) or in FASTA format</span>
            </div>
          </div>
          <div class="control-group">
            <div class="controls">
              <input type="text" name="ncbi" id="ncbi" placeholder="NCBI ACC #">
              <span class="help-inline">Or input NCBI accession number of desired file</span>
            </div>
          </div>
        </fieldset>
        <div id="region-control" data-toggle="collapse" data-target=".region">
          <i class="icon-plus" id="region-button"></i>
          Limit prediction to an input region (ignored for multi-sequence records)
          <a href="#" data-toggle="tooltip" data-placement="right" title="Set start/end coordinates for analysis"><i class="icon-question-sign"></i></a>
        </div>
        <fieldset class="as-form collapse region">
          <div class="controls collapse region">
            <label for="from">Start at
              <input type="text" name="from" id="from" placeholder="start at bp"></label>
          </div>
          <div class="controls collapse region">
            <label for="to">Stop at
              <input type="text" name="to" id="to" placeholder="stop at bp"></label>
          </div>
        </fieldset>
        <div id="cf-control" data-toggle="collapse" data-target=".clusterfinder">
          <i class="icon-plus" id="cf-button"></i>
          Detect putative gene clusters using the <a href="http://www.cell.com/cell/abstract/S0092-8674(14)00826-5">ClusterFinder</a> algorithm
        </div>
        <fieldset class="clusterfinder collapse as-form">
          <div class="control clusterfinder collapse">
            <label class="checkbox" for="inclusive">
              <input type="checkbox" name="inclusive" id="inclusive">
              Enable ClusterFinder algorithm</label>
          </div>
          <div class"control clusterfinder collapse">
            For ClusterFinder-predicted clusters:
          </div>
          <div class="control clusterfinder collapse">
            <label for="cf_cdsnr">Minimum cluster size in CDS
              <input type="text" name="cf_cdsnr" id="cf_cdsnr" placeholder="5"></label>
          </div>
          <div class="control clusterfinder collapse">
            <label for="cf_npfams">Minimum number of biosynthesis-related PFAM domains
              <input type="text" name="cf_npfams" id="cf_npfams" placeholder="5"></label>
          </div>
          <div class="control clusterfinder collapse">
            <label for="cf_threshold">Minimum ClusterFinder probability
              <input type="text" name="cf_threshold" id="cf_threshold" placeholder="0.6"></label>
          </div>
        </fieldset>
        <fieldset class="as-form">
          <div class="control-group">
            <label class="checkbox" for="eukaryotic">
              <input type="checkbox" name="eukaryotic" id="eukaryotic">
              DNA of Eukaryotic origin</label>
          </div>
        </fieldset>
        <fieldset class="control-group as-form dna_related">
          <div>Gene finding options:</div>
          <div class="control">
            <label class="radio inline">
              <input type="radio" name="genefinder" id="prodigal" value="prodigal" checked>
              Prodigal
            </label>
            <label class="radio inline">
              <input type="radio" name="genefinder" id="prodigal_m" value="prodigal_m">
              Prodigal Metagenomic
            </label>
            <label class="radio inline">
              <input type="radio" name="genefinder" id="glimmer" value="glimmer">
              Glimmer
            </label>
          </div>
          <div class="control glimmer">
            <label class="control-label" for="trans_table">GenBank translation table used for Glimmer:</label>
            <select id="trans_table" name="trans_table">
              <option value="1"> The Standard Code</option>
              <option value="11"> The Bacterial, Archaeal and Plant Plastid Code</option>
              <option value="10"> The Euplotid Nuclear Code</option>
              <option value="13"> The Ascidian Mitochondrial Code</option>
              <option value="12"> The Alternative Yeast Nuclear Code</option>
              <option value="15"> Blepharisma Nuclear Code</option>
              <option value="14"> The Alternative Flatworm Mitochondrial Code</option>
              <option value="22"> Scenedesmus Obliquus Mitochondrial Code</option>
              <option value="16"> Chlorophycean Mitochondrial Code</option>
              <option value="23"> Thraustochytrium Mitochondrial Code</option>
              <option value="3"> The Yeast Mitochondrial Code</option>
              <option value="2"> The Vertebrate Mitochondrial Code</option>
              <option value="5"> The Invertebrate Mitochondrial Code</option>
              <option value="4"> The Mold, Protozoan, and Coelenterate Mitochondrial Code and the Mycoplasma/Spiroplasma Code</option>
              <option value="6"> The Ciliate, Dasycladacean and Hexamita Nuclear Code</option>
              <option value="9"> The Echinoderm and Flatworm Mitochondrial Code</option>
              <option value="21"> Trematode Mitochondrial Code</option>
            </select>
          </div>
          <div class="control glimmer">
            <input type="text" name="gene_length" id="gene_length" value="50">
            <span class="help-inline">Glimmer minimal gene length</span>
          </div>
          <div class="control">
            <label class="checkbox" for="all_orfs">
              <input type="checkbox" name="all_orfs" id="all_orfs" checked>
              Run secondary metabolite detection on all possible ORFs</label>
            </div>

        </fieldset>
        <div>
          <strong>BLAST comparisons to other gene clusters:</strong>
        </div>
        <fieldset class="control-group as-form">
          <div class="control span4">
            <label class="checkbox" for="clusterblast">
              <input type="checkbox" name="clusterblast" id="clusterblast" checked>
              Gene Cluster Blast analysis</label>
          </div>
          <div class="control span4">
            <label class="checkbox" for="knownclusterblast">
              <input type="checkbox" name="knownclusterblast" id="knownclusterblast" checked>
              Known Gene Cluster Blast analysis</label>
          </div>
          <div class="control span4">
            <label class="checkbox" for="subclusterblast">
              <input type="checkbox" name="subclusterblast" id="subclusterblast" checked>
              Subcluster Blast analysis</label>
          </div>
        </fieldset>
        <div>
          <strong>Additional annotations:</strong>
        </div>
        <fieldset class="control-group as-form">
          <div class="control span4">
            <label class="checkbox" for="smcogs">
              <input type="checkbox" name="smcogs" id="smcogs" checked>
              smCOG analysis for functional prediction and phylogenetic analysis of genes</label>
          </div>
          <div class="control span4">
            <label class="checkbox" for="asf">
              <input type="checkbox" name="asf" id="asf" checked>
              Active site finder</label>
          </div>
        </fieldset>
        <div>
          <strong>Optional analyses with a long runtime</strong>
        </div>
        <fieldset class="control-group">
          <div class="control">
            <label class="checkbox" for="fullhmmer">
              <input type="checkbox" name="fullhmmer" id="fullhmmer">
              Whole-genome PFAM analysis</label>
          </div>
        </fieldset>
        <fieldset class="control-group">
          <div class="span6">
            Homology based metabolic modeling pipeline against template model:
          </div>
          <div class="control span6">
            <label class="radio inline">
              <input type="radio" name="modeling" id="modeling_none" value="none" checked>
              none</label>
            <label class="radio inline">
              <input type="radio" name="modeling" id="modeling_eco" value="eco">
              <i>E. coli</i></label>
            <label class="radio inline">
              <input type="radio" name="modeling" id="modeling_sco" value="sco">
              <i>S. coelicolor</i></label>
          </div>
        </fieldset>
        <fieldset class=control-group">
          <div class="control">
            <label class="checkbox" for="ecpred">
              <input type="checkbox" name="ecpred" id="ecpred">
              Enzyme Commission (EC) number prediction (very long runtime)</label>
          </div>
        </fieldset>
        <div class="form-actions">
          <button type="submit" class="btn btn-primary">Submit</button>
        </div>
      </form>
    </div>
  </div>
  <div class="tab-pane" id="prot">
    <div>Analyze one or more amino acid sequences to identify secondary metabolite scaffold biosynthesis genes:</div>
    <div>
      <div class="control-group">
        <button class="btn btn-inverse" id="prot-sample-input">Load sample input</button>
        <button class="btn btn-inverse" id="prot-sample-output">Open example output</button>
      </div>
      <form action="{{ url_for('protein') }}" id="prot-form" method="POST">
        <div class="control-group">
          <div class="controls">
            <input type="email" name="email" id="email" value="{{old_email}}" placeholder="your.email@example.com">
            <span class="help-inline">Email address (optional)</span>
          </div>
          <div class="controls">
            <input type="text" name="prot-ncbi" id="prot-ncbi" placeholder="NCBI ACC #">
            <span class="help-inline">Input (comma-separated) NCBI accession number(s) of desired protein(s)</span>
          </div>
          <div class="controls">
            <label for="sequence">or paste a FASTA amino acid sequence here:</label>
            <textarea name="sequence" id="sequence" rows="10" class="input-block-level" >{{old_sequence}}</textarea>
          </div>
        </div>
        <div class="form-actions">
          <button type="submit" class="btn btn-primary">Submit</button>
        </div>
      </form>
    </div>
  </div>
  <div class="tab-pane" id="job">
    <div>Display results for an existing job</div>
    <form class="form-inline">
      <label for="jobid">Job ID:
        <input type="text" name="jobid" id="jobid" class="input-xlarge" placeholder="aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee">
      </label>
      <button id="display-job" class="btn btn-primary">Display</button>
    </form>
  </div>
</div>
{% endblock %}
{% block ready_function %}
    // Set up button actions
    $('#dna-sample-input').click(function(){
        $('#ncbi').val('Y16952');
        clear_upload();
    });
    $('#dna-sample-output').click(function(){
        window.open('/upload/example/index.html', '_self');
    });
    $('#prot-sample-input').click(function(){
        $('#prot-ncbi').val('ADJ56352.1,ADJ56353.1');
        clear_sequence();
    });
    $('#prot-sample-output').click(function(){
        window.open('/upload/prot_example/index.html', '_self');
    });

    // Allow the 'all' checkbox to toggle all other clusters
    $('#cluster_1').change(toggle_clusters);
    $('#cluster_1').prop('checked', 'checked');
    toggle_clusters();

    // disable new options for legacy antismash1 jobs
    $('#legacy').change(toggle_legacy);
    toggle_legacy();

    // Show genefinding options if user is trying to upload a fasta file
    $('#eukaryotic').change(show_genefinding_if_needed);
    $('input[name=genefinder]').change(show_genefinding_if_needed);

    // Make sure that only NCBI accession# or uploadable file is present.
    $('#seq').change(seq_callback);
    $('#ncbi').change(clear_upload);

    $('#sequence').change(clear_prot_ncbi);
    $('#prot-ncbi').change(clear_sequence);

    show_genefinding_if_needed();

    $('.legacy-options').on('shown', function(){
        $('#legacy-button').removeClass('icon-plus')
            .addClass('icon-minus');
        $('#legacy-description').text('Hide');
    })
    .on('hidden', function() {
        $('#legacy-button').removeClass('icon-minus')
            .addClass('icon-plus');
        $('#legacy-description').text('Show');
    });

    $('.region').on('shown', function(){
        $('#region-button').removeClass('icon-plus')
            .addClass('icon-minus');
    })
    .on('hidden', function() {
        $('#region-button').removeClass('icon-minus')
            .addClass('icon-plus');
    });

    $('.sec-met').on('shown', function(){
        $('#sec-met-button').removeClass('icon-plus')
            .addClass('icon-minus');
    })
    .on('hidden', function() {
        $('#sec-met-button').removeClass('icon-minus')
            .addClass('icon-plus');
    });
    $('a[data-toggle="tooltip"]').tooltip();

    $('#display-job').click(function(){
        var jobid = $('#jobid').val();
        window.open("{{ url_for('display', task_id='') }}" + jobid, "_self");
        return false;
    });

    {% if switch_to %}
    $('.nav-tabs a[href="#{{switch_to}}"]').tab('show');
    {% endif %}

    $('#nucl-form').submit(verify_nucl_form);
    $('#prot-form').submit(verify_prot_form);
{% endblock %}
